tests/unit/test_sap_rfc.py

```diff
@@ -0,0 +1,190 @@
+import pandas as pd
+
+
+@pytest.fixture(scope="session")
+def sap():
+    sap = SAPRFC(config_key="sap_rfc")
+    yield sap
+
+
+sql1 = "SELECT a AS a_renamed, b FROM table1 WHERE table1.c = 1"
+sql2 = "SELECT a FROM fake_schema.fake_table WHERE a=1 AND b=2 OR c LIKE 'a%' AND d IN (1, 2) LIMIT 5 OFFSET 3"
+sql3 = "SELECT b FROM c WHERE testORword=1 AND testANDword=2 AND testLIMITword=3 AND testOFFSETword=4"
+sql4 = "SELECT c FROM d WHERE testLIMIT = 1 AND testOFFSET = 2 AND LIMITtest=3 AND OFFSETtest=4"
+sql5 = sql3 + " AND longword123=5"
+sql6 = "SELECT a FROM fake_schema.fake_table WHERE a=1 AND b=2 OR c LIKE 'a%' AND d IN (1, 2) AND longcolname=3 AND otherlongcolname=5 LIMIT 5 OFFSET 3"
+sql7 = """
+SELECT a, b
+FROM b
+WHERE c = 1
+AND d = 2
+AND longcolname = 12345
+AND otherlongcolname = 6789
+AND thirdlongcolname = 01234
+LIMIT 5
+OFFSET 10
+"""
+sql8 = "INSERT INTO customers name VALUES 'John Doe'"
+sql9 = " SELECT * FROM table1"
+sql10 = "SELECT a FROM fake_schema.fake_table WHERE a=1 AND b=2 OR c LIKE 'a%' AND d IN (1, 2) AND longcolname=3 OR otherlongcolname=5 LIMIT 5 OFFSET 3"
+sql11 = "SELECT col FROM table1, table2"
+
+
+def test_get_function_parameters(sap):
+    assert sap.get_function_parameters("RFC_READ_TABLE", description=None) == [
+        "ET_DATA",
+        "DELIMITER",
+        "GET_SORTED",
+        "NO_DATA",
+        "QUERY_TABLE",
+        "ROWCOUNT",
+        "ROWSKIPS",
+        "USE_ET_DATA_4_RETURN",
+        "DATA",
+        "FIELDS",
+        "OPTIONS",
+    ]
+    assert (
+        pd.DataFrame.to_json(sap.get_function_parameters("RFC_READ_TABLE", "short"))
+        == '{"name":{"0":"ET_DATA","1":"DELIMITER","2":"GET_SORTED","3":"NO_DATA","4":"QUERY_TABLE","5":"ROWCOUNT","6":"ROWSKIPS","7":"USE_ET_DATA_4_RETURN","8":"DATA","9":"FIELDS","10":"OPTIONS"},"parameter_type":{"0":"RFCTYPE_TABLE","1":"RFCTYPE_CHAR","2":"RFCTYPE_CHAR","3":"RFCTYPE_CHAR","4":"RFCTYPE_CHAR","5":"RFCTYPE_INT","6":"RFCTYPE_INT","7":"RFCTYPE_CHAR","8":"RFCTYPE_TABLE","9":"RFCTYPE_TABLE","10":"RFCTYPE_TABLE"},"default_value":{"0":"","1":"SPACE","2":"","3":"SPACE","4":"","5":"0","6":"0","7":"","8":"","9":"","10":""},"optional":{"0":false,"1":true,"2":true,"3":true,"4":false,"5":true,"6":true,"7":true,"8":true,"9":true,"10":true},"parameter_text":{"0":"","1":"Sign for indicating field limits in DATA","2":"","3":"If <> SPACE, only FIELDS is filled","4":"Table read","5":"","6":"","7":"","8":"Data read (out)","9":"Names (in) and structure (out) of fields read","10":"Selection entries, \\"WHERE clauses\\" (in)"}}'
+    )
+    with pytest.raises(
+        ValueError,
+        match="Incorrect value for 'description'. Correct values",
+    ):
+        sap.get_function_parameters("RFC_READ_TABLE", "x")
+
+
+def test__get_table_name(sap):
+    assert sap._get_table_name(sql1) == "table1"
+    assert sap._get_table_name(sql2) == "fake_schema.fake_table", sap._get_table_name(
+        sql2
+    )
+    assert sap._get_table_name(sql3) == "c"
```

Comment:
Awkward. Need more explanations about use-cases there are gona to be tested and need human readable sql-variables naming